stdlib/public/core/Sort.swift.gyb

```diff
@@ -72,58 +72,105 @@ func _insertionSort<C>(
   }
 }
 
+/// Sorts the elements at indices `a`, `b`, and `c`. Stable.
+///
+/// - Precondition: `a < b && b < c`
+/// - Postcondition: `elements[a] <= elements[b] && elements[b] <= elements[c]`
+public // @testable
+func _sort3<C>(
+  _ elements: inout C,
+  _ a: C.Index, _ b: C.Index, _ c: C.Index
+  ${", by areInIncreasingOrder: (C.Iterator.Element, C.Iterator.Element) -> Bool" if p else ""}
+)
```

Comment:
I'm following the rest of the functions in the file, here. It would be great to have all of these as extension methods and to get rid of the gyb—I can try those changes out after we get this merged.